File: 2561-number-of-distinct-averages/2561-number-of-distinct-averages.java
class Solution {
    public int distinctAverages(int[] nums) {
        Arrays.sort(nums);
        List<Double> res = new ArrayList<>();
        int l = 0 , r = nums.length - 1;
        while(l < r){
            double num = (double) (nums[l] + nums[r]) / 2;
            //System.out.print(num);
            if(!res.contains(num)){
                res.add(num);
            }
            l++;
            r--;
        }
        return res.size();
    }
}

File: 2561-number-of-distinct-averages/2561-number-of-distinct-averages.py
class Solution:
    def distinctAverages(self, nums: List[int]) -> int:
        nums.sort()
        res = []
        l , r = 0 , len(nums) - 1
        while l < r :
            num = (nums[l] + nums[r]) / 2
            if num not in res:
                res.append(num)
            l += 1
            r -= 1
        return len(res)